main.tf

```
@@ -62,11 +63,16 @@ locals {
     module.sso_roles.by_name[name]
   ]
 
-  secret_principals = concat(
+  secret_write_principals = concat(
     local.execution_role_arns,
     local.secret_permission_set_roles
   )
 
+  secret_read_principals = concat(
+    local.secret_write_principals,
```

Should it be read instead?

```suggestion
    local.secret_read_principals,
```